db/corruption_test.cc

```diff
@@ -1131,25 +1131,32 @@ TEST_P(CrashDuringRecoveryWithCorruptionTest, CrashDuringRecovery) {
   // Case2: If avoid_flush_during_recovery = false, all column families have
   // flushed their data from WAL to L0 during recovery, and none of them will
   // ever need to read the WALs again.
+
+  // 4. Fault is injected which emulates second power reset causing
```

Comment:
I don't think we are emulating losing unsync'ed WAL here. The reason we inject an error here is to let recovery fail and expose the bug of prematurely persisting a new MANIFEST with advanced `ColumnFamilyData::log_number`.
Truncating the file is emulating losing unsynced WAL.

Reply:
Oh ok. so we are addressing two different issues.

Reply:
We are still addressing the original bug of incorrectly persisting a new MANIFEST. The combination of corrupting a WAL and injecting an error during this re-open aims to reproduce the same bug. Let me know if I miss anything.